app/assets/javascripts/annotations.js

```diff
@@ -745,7 +782,11 @@ function newAnnotationBox(annotation) {
     const problem_id = $(this).val();
 
     // Update autocomplete to display shared comments for selected problem
-    box.find("#comment-textarea").autocomplete("option", "source", localCache["shared_comments"][problem_id] || []);
+    box.find("#comment-textarea").autocomplete({
+        source: getSharedCommentsForProblem(problem_id) || [],
```

Comment:
Nit: Could we use a consistent ordering for the source, select, and focus options?